lib/navigation.js

```diff
@@ -0,0 +1,51 @@
+// Navgation metalsmith plugin
+// builds a navigation object based on config file and folder structure
+
+// files: array containing information about every page
+// metalsmith: object containing global information such as meta data
+// done: function which must be called when the plugin has finished working
+
+const fs = require('fs')
+const path = require('path')
+const paths = require('../config/paths.json')
+const navigation = require('../config/navigation.json')
+
+module.exports = function () {
+  return function (files, metalsmith, done) {
+    // iterate main navigation item defined in navigation.json
+    for (let item in navigation) {
+      // if is not homepage look for subitems
+      if (navigation[item].url !== '') {
+        // define source path
+        let itemPath = paths.source + navigation[item].url
+        // get directories under main navigation item (e.g. ['breadcrumbs', 'checkboxes', ...])
+        let directories = fs.readdirSync(itemPath).filter(filePath => fs.statSync(path.join(itemPath, filePath)).isDirectory())
```

Comment:
This works well but it also looks like this line is doing an awful lot 🙂   It might be good to consider moving this work into a function, also to potentially help with debugging as it'll be easier to see which part of the line has failed and for adding some comments to explain what the code is doing.

Reply:
agree with hanna - breaking this up into steps would make it more readable

Reply:
I'll split this line out as discussed yesterday.

Reply:
Updated, moved this as a function in `file-helper` with proper try-catch on file reading.